content/blog/2022-06-07-using-gcp-remotes-in-dvc.md

```diff
@@ -0,0 +1,158 @@
+
+The Bucket page will have a lot of configurations you can set, but you can leave
+the settings in the default state if there’s nothing you need to customize. We
+have named this example bucket `updatedbikedata` as you can see below.
+
+![gcp_bucket_options.png](/uploads/images/2022-06-07/gcp_bucket_options.png)
+
+Now you can save your changes and you’ll be redirected to the `Bucket Details`
+page and you’ll see the bucket you just created.
+
+![created_gcp_bucket.png](/uploads/images/2022-06-07/created_gcp_bucket.png)
+
+### Get your credentials
+
+Since you have the bucket created, we need to get the credentials to connect the
+GCP remote to the project. Go to the `IAM & Admin` service and go to
+`Service Accounts` in the left sidebar. You’ll see your project and you’ll be
+able to click on `Actions` and go to where you `Manage keys` for your project.
+
+![gcp_service_account.png](/uploads/images/2022-06-07/gcp_service_account.png)
+
+Once you’ve been redirected, click the `Add Key` button and this will bring up
+the credentials you need to authenticate your GCP account with your project. Go
+ahead and download the credentials file and store it somewhere safe.
+
+![gcp_key.png](/uploads/images/2022-06-07/gcp_key.png)
+
+That’s it for setting up your storage bucket and getting the credentials you
+need! Now let’s add DVC to our demo repo and set up the remote.
+
+## Set up a DVC project
+
+First, add DVC as a requirement to your project with the following installation
+command:
+
+`$ pip install 'dvc[gs]'`
+
+Then you can initialize DVC in your own project with the following command:
+
+`$ dvc init`
+
+This will add all of the DVC internals needed to start versioning your data and
+tracking experiments. Now we need to set up the remote to connect our project
+data stored in GCP to the DVC repo.
+
+### Create a default remote
+
+Now we can make the GCP storage the default remote for the project with the
+following command:
+
+`$ dvc remote add -d bikes gs://updatedbikedata`
+
+This creates a default remote called `bikes` that connects to the
+`updatedbikedata` bucket we made earlier which is where the any data for the
+model will be stored.
+
+### Add GCP credentials
+
+In order for DVC to be able to push and pull data from the remote, you need to
+have valid AWS credentials.
+
+If you are using the GCP CLI already, you should be able to run
+`gcloud auth application-default login`. You can also authenticate with the
+service account we created earlier in a couple of ways with that credentials
+file we downloaded.
+
+You can set the `GOOGLE_APPLICATION_CREDENTIALS` environment variable to point
+to the credentials file, like this:
+
+```dvc
+$ export GOOGLE_APPLICATION_CREDENTIALS='../tonal-history-154018-e62a79baf90f.json'
+```
+
+Or you can add the credentials file location with the following command:
+
+```dvc
+$ dvc remote modify bikes credentialpath '../tonal-history-154018-e62a79baf90f.json'
```

Comment:
Should we suggest using `--local` since others are unlikely to have the same path?

Reply:
That makes sense to me. Thanks for catching that!